lib/common/util.js

```diff
@@ -545,3 +545,43 @@ function normalizeArguments(globalContext, localConfig, options) {
 }
 
 util.normalizeArguments = normalizeArguments;
+
+/**
+ * Throttle requests according to a `maxApiCalls` limit.
+ *
+ * @param {function} makeRequestFn - The function that will be called.
+ * @param {object=} options - Configuration object.
+ * @param {number} options.maxApiCalls - The maximum number of API calls to
+ *     make.
+ */
+function iterator(makeRequestFn, options) {
```

Comment:
Sure, changed. The idea was that, like a generator/iterator, it requires an explicit "next()" that triggers "doing the next thing". In this case, that's "makeRequest". Just words, though. Whatever makes sense to you.